### PR TITLE
CVar mp_fadetoblack enhancement (close #463)

### LINKED ISSUE #463: [feature request] add screenfade on kill like into CS:GO
effect like in CS:GO
https://www.youtube.com/watch?v=Gp9CR-xEKTw

### PR DESCRIPTION
**Pull request by wopox1337**
closes https://github.com/s1lentq/ReGameDLL_CS/issues/463
```
// Observer's screen will fade to black on kill event or permanent.
// 0 - No fade
// 1 - Fade to black and won't be able to watch anybody
// 2 - fade to black only on kill moment.
//
// Default value: "0"
mp_fadetoblack 0
```

https://youtu.be/OsLB_mg5hYg

### CODE REVIEW

**Comment by wopox1337** [comment]
First problem.  When you die blinded by a flash of black covers the white. 
https://youtu.be/YZ69uOJtLBU

File: regamedll/dlls/player.cpp [hunk 1]
```diff
@@ -2078,6 +2079,53 @@ void EXT_FUNC CBasePlayer::__API_HOOK(Killed)(entvars_t *pevAttacker, int iGib)
...
+		m_hObserverTarget = UTIL_PlayerByIndexSafe(pev->iuser3);
+
+		MESSAGE_BEGIN(MSG_ONE, gmsgADStop, nullptr, pev);
+		MESSAGE_END();
+
+		for (int i = 1; i <= gpGlobals->maxClients; i++)
+		{
+			CBasePlayer* pObserver = UTIL_PlayerByIndex(i);
+
+			if (pObserver == this || pObserver && pObserver->IsObservingPlayer(this))
```
[the remark this answers is not in the document]
Comment (wopox1337) [reply regamedll/dlls/player.cpp]: IF you have any IDEAS how to make the code better - I am ready to accept them

[the remark this answers is not in the document]
Comment (Vaqtincha) [reply regamedll/dlls/player.cpp]: Если все еще непонятно: тут нужны скобки (где указать не буду)

File: regamedll/dlls/player.cpp [hunk 2]
```diff
@@ -2078,6 +2079,53 @@ void EXT_FUNC CBasePlayer::__API_HOOK(Killed)(entvars_t *pevAttacker, int iGib)
...
+			{
+				UTIL_ScreenFade(pObserver, Vector(0, 0, 0), 1, 4, 255, (FFADE_OUT));
```
Comment (wopox1337) [inline regamedll/dlls/player.cpp]: Wait https://github.com/s1lentq/ReGameDLL_CS/pull/483 PR 
need to change timing for new CVar.

File: regamedll/dlls/player.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (wopox1337) [reply regamedll/dlls/player.cpp]: I suggest you read some documentation
https://www.tutorialspoint.com/Operators-Precedence-in-Cplusplus

**Pull request merged**